package com.becoder.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@EntityListeners(AuditingEntityListener.class)
public class Notes extends BaseModel{

      @Id
      @GeneratedValue(strategy = GenerationType.IDENTITY)
      private Integer id;

      private String title;

      private String description;

      @ManyToOne
      private Category category;

      @ManyToOne
      private FileDetails fileDetails;

      private boolean isDeleted;

      private LocalDateTime deletedOn;



}
